backend/src/services/course/course.service.ts

```diff
@@ -24,6 +24,27 @@ class Course {
     this.#udemyService = udemyService;
   }
 
+  async getAll(): Promise<CourseGetResponseDto[]> {
+    const courses = await this.#courseRepository.getAll();
+
+    return Promise.all(
+      courses.map(async (course) => {
+        const vendor = await this.#vendorService.getById(course.vendorId);
+
+        if (!vendor) {
+          throw new CoursesError({
+            message: ExceptionMessage.INVALID_COURSE_VENDOR,
+          });
+        }
+
+        return {
+          ...course,
+          vendorKey: vendor.key as VendorKey,
+        };
+      }),
+    );
```

Comment:
We can get vendors linked to each course by declaring relation between `course` and `vendor` models